azurerm/internal/services/mssql/resource_arm_mssql_virtual_machine.go

```diff
@@ -0,0 +1,503 @@
+			"virtual_machine_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"sql_license_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(sqlvirtualmachine.PAYG),
+					string(sqlvirtualmachine.AHUB),
+				}, false),
+			},
+
+			"sql_virtual_machine_group_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"auto_patching": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"day_of_week": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(sqlvirtualmachine.Monday),
+								string(sqlvirtualmachine.Tuesday),
+								string(sqlvirtualmachine.Wednesday),
+								string(sqlvirtualmachine.Thursday),
+								string(sqlvirtualmachine.Friday),
+								string(sqlvirtualmachine.Saturday),
+								string(sqlvirtualmachine.Sunday),
+							}, false),
+						},
+
+						"maintenance_window_duration_in_minutes": {
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntBetween(30, 180),
+						},
+
+						"maintenance_window_starting_hour": {
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntBetween(0, 23),
+						},
+					},
+				},
+			},
+
+			"key_vault_credential": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validate.NoEmptyStrings,
+							//api will add updated credential name, and return "sqlvmName:name1,sqlvmName:name2"
+							DiffSuppressFunc: mssqlVMCredentialNameDiffSuppressFunc,
+						},
+
+						"azure_key_vault_url": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							Sensitive:    true,
+							ValidateFunc: validate.URLIsHTTPS,
+						},
+
+						"service_principal_name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							Sensitive:    true,
+							ValidateFunc: validate.NoEmptyStrings,
+						},
+
+						"service_principal_secret": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							Sensitive:    true,
+							ValidateFunc: validate.NoEmptyStrings,
+						},
+					},
+				},
+			},
+
+			"server_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"is_r_services_enabled": {
```

Comment:
could we remove the is here
```suggestion
						"r_services_enabled": {
```